tools/importer-msgraph-metadata/pipeline/task_parse_resources.go

```diff
@@ -130,26 +123,38 @@ func (p pipelineTask) parseResourcesForService(resourceIds ResourceIds, models M
 					}
 
 					if resp.Value != nil && len(resp.Value.Content) > 0 {
+						if resp.Value.Description != nil && strings.Contains(strings.ToLower(*resp.Value.Description), "collection") {
+							listOperation = true
+						}
 						for t, m := range resp.Value.Content {
 							contentType = &t
 
+							if strings.HasPrefix(strings.ToLower(t), "text/plain") {
+								continue
+							}
+
+							// Prefer model name from Ref
 							if strings.HasPrefix(m.Schema.Ref, refPrefix) {
 								modelName := cleanName(m.Schema.Ref[len(refPrefix):])
 								responseModel = &modelName
-							} else if m.Schema != nil {
+							}
+
+							if m.Schema != nil {
+								// Flatten the response SchemaRef for inspection
 								if f, _ := flattenSchemaRef(m.Schema, nil); f != nil {
 									if f.Format == "binary" {
 										responseType = pointerTo(DataTypeBinary)
 										break
 									}
 
+									// Derive model and response type from title
 									if title := f.Title; title != "" || f.Type != "" {
-										if strings.HasPrefix(strings.ToLower(title), "collection of ") {
-											title = title[14:]
+										if strings.HasPrefix(title, "CollectionOf") {
```

Comment:
pedantic, but technically should this be insensitive given the input source?

Reply:
good shout 👍